src/model/networking/connection/callback/io-callbacks.h

```diff
@@ -0,0 +1,39 @@
+#ifndef MODEL_NETWORKING_CONNECTION_CALLBACK_IO_H_
+#define MODEL_NETWORKING_CONNECTION_CALLBACK_IO_H_
+
+#include <event2/bufferevent.h>
+#include <event2/event.h>
+
+#include "model/networking/connection/connection.h"
+
+namespace model {
+class IOCallbacks {
+public:
+    static void ReadMessageClientCbHandler(
+        struct bufferevent* bev,
+        void* ptr);
+
+    static void ReadMessageServerCbHandler(
+        struct bufferevent* bev,
+        void* ptr);
+
+    static void WriteMessageCbHandler(
+        struct bufferevent* bev,
+        void* ptr);
+
+    static void EventCbHandler(
+        struct bufferevent* bev,
+        short events,
+        void* ptr);
+
+    static void SetClientConnectionCallbacks(
+        struct bufferevent* bev,
+        Connection* connection);
+
+    static void SetServerConnectionCallbacks(
+        struct bufferevent* bev,
+        Connection* connection);
+};
+} // namespace model
```

Comment:
Quite a few months ago I had them as functions because it did not make sense to keep them in a class. I'm not sure why but the application didn't like it at all and caused all sorts of problems so I changed it back to using methods. I would prefer to use standard functions but I think there is some strange cpp thing happening. I will give it another go as its been a long time now.